### ISSUE TITLE
Cannot load location

### ISSUE THREAD

**Issue by bushvin** [cyberjunky/home-assistant-shell_recharge#11]
## System Information

version | core-2024.1.5
-- | --
installation_type | Home Assistant OS
dev | false
hassio | true
docker | true
user | root
virtualenv | false
python_version | 3.11.6
os_name | Linux
os_version | 6.1.63-haos-raspi
arch | aarch64
timezone | XXXXX
config_dir | /config

<details><summary>Home Assistant Community Store</summary>

GitHub API | ok
-- | --
GitHub Content | ok
GitHub Web | ok
GitHub API Calls Remaining | 4997
Installed Version | 1.34.0
Stage | running
Available Repositories | 1392
Downloaded Repositories | 24

</details>

<details><summary>Home Assistant Cloud</summary>

logged_in | false
-- | --
can_reach_cert_server | ok
can_reach_cloud_auth | ok
can_reach_cloud | ok

</details>

<details><summary>Home Assistant Supervisor</summary>

host_os | Home Assistant OS 11.4
-- | --
update_channel | stable
supervisor_version | supervisor-2023.12.1
agent_version | 1.6.0
docker_version | 24.0.7
disk_total | 879.5 GB
disk_used | 105.2 GB
healthy | true
supported | true
board | rpi4-64
supervisor_api | ok
version_api | ok
installed_addons | N/A

</details>

<details><summary>Dashboards</summary>

dashboards | XX
-- | --
resources | XX
views | XX
mode | XXX

</details>

<details><summary>Recorder</summary>

oldest_recorder_run | February 1, 2023 at 07:08
-- | --
current_recorder_run | January 29, 2024 at 12:14
estimated_db_size | XXX
database_engine | XXX
database_version | XXX

</details>

### Checklist

- [X] I have enabled debug logging for my installation.
- [X] I have filled out the issue template to the best of my ability.
- [X] This issue only contains 1 issue (if you have multiple issues, open one issue for each issue).
- [X] This issue is not a duplicate issue of any [previous issues](https://github.com/cyberjunky/home-assistant-shell_recharge/issues?q=is%3Aissue+label%3A%22Bug%22+)..

### Describe the issue

When providing 682154 as a serial number, no information is retrieved and the message "An Unknown Error has ocurred" Appears.

After some investigation, it seems there might be an issue with certain properties missing from the endpoint.
I checked out your shellrecharge library code, and discovered 'my' EV does indeed exist, but after some digging, it might be related to your Tariff Model, as it looks for `perKWh`, but this one only seems to support `perMinute`

This is the API endpoint which craps out:
https://ui-map.shellrecharge.com/api/map/v2/locations/search/682154

This is the address for the EV with the abovementioned S/N: Esplanade des Islettes, 73450, Valmeinier

### Reproduction steps

1. Go to settings >> Devices and Services
2. Click Add Integration and enter "682154" (without quotes) as the serial number
3. bask in red light of the error which comes up.
...

### Debug logs

```text
2024-01-29 13:53:33.501 ERROR (MainThread) [shellrecharge] 7 validation errors for Location
evses -> 0 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 0 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
openTwentyFourSeven
  field required (type=value_error.missing)
openingHours
  field required (type=value_error.missing)
predictedOccupancies
  field required (type=value_error.missing)
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/shellrecharge/__init__.py", line 41, in location_by_id
    location_data = Location.parse_obj(result[0])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 526, in parse_obj
    return cls(**obj)
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 341, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 7 validation errors for Location
evses -> 0 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 0 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
openTwentyFourSeven
  field required (type=value_error.missing)
openingHours
  field required (type=value_error.missing)
predictedOccupancies
  field required (type=value_error.missing)
2024-01-29 13:53:33.512 ERROR (MainThread) [aiohttp.server] Error handling request
Traceback (most recent call last):
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_protocol.py", line 452, in _handle_request
    resp = await request_handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_app.py", line 543, in _handle
    resp = await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/aiohttp/web_middlewares.py", line 114, in impl
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/security_filter.py", line 85, in security_filter_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/forwarded.py", line 227, in forwarded_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/request_context.py", line 28, in request_context_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/ban.py", line 80, in ban_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/auth.py", line 233, in auth_middleware
    return await handler(request)
           ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/headers.py", line 31, in headers_middleware
    response = await handler(request)
               ^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/view.py", line 149, in handle
    result = await handler(request, **request.match_info)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/decorators.py", line 63, in with_admin
    return await func(self, request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/config/config_entries.py", line 177, in post
    return await super().post(request, flow_id)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/components/http/data_validator.py", line 72, in wrapper
    result = await method(view, request, data, *args, **kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/helpers/data_entry_flow.py", line 110, in post
    result = await self._flow_mgr.async_configure(flow_id, data)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 320, in async_configure
    result = await self._async_handle_step(
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/src/homeassistant/homeassistant/data_entry_flow.py", line 416, in _async_handle_step
    result: FlowResult = await getattr(flow, method)(user_input)
                         ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/config/custom_components/shell_recharge/config_flow.py", line 34, in async_step_user
    if not await api.location_by_id(user_input["serial_number"]):
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/shellrecharge/__init__.py", line 50, in location_by_id
    raise err
  File "/usr/local/lib/python3.11/site-packages/shellrecharge/__init__.py", line 41, in location_by_id
    location_data = Location.parse_obj(result[0])
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 526, in parse_obj
    return cls(**obj)
           ^^^^^^^^^^
  File "/usr/local/lib/python3.11/site-packages/pydantic/main.py", line 341, in __init__
    raise validation_error
pydantic.error_wrappers.ValidationError: 7 validation errors for Location
evses -> 0 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 0 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 0 -> tariff -> perKWh
  field required (type=value_error.missing)
evses -> 1 -> connectors -> 1 -> tariff -> perKWh
  field required (type=value_error.missing)
openTwentyFourSeven
  field required (type=value_error.missing)
openingHours
  field required (type=value_error.missing)
predictedOccupancies
  field required (type=value_error.missing)
```

**Comment by cyberjunky** [comment]
Thank you for reporting, I will check/fix it, this integration is very new and still in development ;-) But bug reports are very useful! I noticed the Shell recharge data can vary a lot...

**Comment by cyberjunky** [comment]
I have fixed it in version 0.1.7 together with better error checking.

**Comment by bushvin** [comment]
Works like a charm! Thanks for fixing this!
